components/scream/src/physics/share/common_physics_impl.hpp

```diff
@@ -0,0 +1,165 @@
+#ifndef COMMON_PHYSICS_IMPL_HPP
+#define COMMON_PHYSICS_IMPL_HPP
+
+#include "common_physics_functions.hpp"
+#include "physics_constants.hpp"
+
+namespace scream {
+namespace physics {
+
+  
+//-----------------------------------------------------------------------------------------------//
+// Applies Exners Function which follows:
+//   Exner = (P/P0)^(Rd/Cp),
+// where,
+//   P  is the pressure at this location, Pa
+//   P0 is a reference pressure, Pa
+//   Rd is the gas constant, J/K
+//   Cp is heat capacity of dry air, J/Ki
+// All universal constants, P0, Rd, and Cp, are defined in physics_constants.hpp
+// Note: Another experssion for Exner is,
+//   Exner = T/th
+// whre,
+//   T  is the temperature, K
+//   th is the potential temperature, K
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_FUNCTION
+ScalarT PhysicsFunctions<DeviceT>::get_exner(const ScalarT& p_mid)
+{
+  using C = scream::physics::Constants<ScalarT>;
+  static constexpr ScalarT p0(C::P0);
+  static constexpr ScalarT rd(C::RD);
+  static constexpr ScalarT inv_cp(C::INV_CP);
+
+  ScalarT exner = pow( p_mid/p0, rd*inv_cp );
+  // Simple check to make sure no obvious errors occurred
+  EKAT_REQUIRE_MSG(!isnan(exner),"Error in get_exner, exner(p = " + std::to_string(p_mid) + ") = NaN"); 
```

Comment:
Also, `isnan(exner)` will return a mask in case of packs. So the check here is tricky. What you wrote works for Real but not for Pack, but `isnan(exner).any()` works for Pack but not for Real.

We could either remove any check inside the fcns (and leaving the responsibility of checking input to the user _before_ calling the physics function), or we should add some utilities in ekat, that work with both Mask and bool. Something like

```
template<typename C>
bool any (const C& c);
template<typename C>
bool none (const C& c);
template<typename C>
bool all (const C& c);
```
which we specialize for `C=Mask` and `C=bool`.

Reply:
I'd vote for the EKAT option.  It may be that in this case we can just move the checks to outside the function, but I can forsee cases in the future where having in-function checks would be desirable.  Should I go ahead and work on that in EKAT?